### PR TITLE
SNOW-1654536: async binding stage bind upload agent

### PR DESCRIPTION
**Pull request by sfc-gh-aling**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes #NNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am modifying authorization mechanisms
   - [ ] I am adding new credentials
   - [ ] I am modifying OCSP code
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.

   Please write a short description of how your code change solves the related issue.

4. (Optional) PR for stored-proc connector:

### CODE REVIEW

File: src/snowflake/connector/aio/_build_upload_agent.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        self,
+        cursor: SnowflakeCursor,
+        rows: list[bytes],
+        stream_buffer_size: int = 1024 * 1024 * 10,
+    ) -> None:
+        super().__init__(cursor, rows, stream_buffer_size)
+        self.cursor = cast("SnowflakeCursor", cursor)
+
+    async def _create_stage(self) -> None:
+        await self.cursor.execute(self._CREATE_STAGE_STMT)
```
Comment (sfc-gh-yuwang) [inline src/snowflake/connector/aio/_build_upload_agent.py:33]: should we log a jira to use scoped temp object since we will add it on sync connector?

Comment (sfc-gh-aling) [reply src/snowflake/connector/aio/_build_upload_agent.py:33]: yes, I have a JIRA to track syncing all main branches into our aio connector: [intermediate](https://snowflakecomputing.atlassian.net/browse/SNOW-1664063)

**Comment by sfc-gh-yuwang** [comment]
I found that old tests all execute query in such way: `connection.cursor().execute`, do you thing change them to `cursor.execute` without creating new cursors each time will make merge gate faster or will it reduce internet overhead?

**Comment by sfc-gh-aling** [comment]
@sfc-gh-yuwang this is a good feedback.

I don't think there is much performance gain here because `cnx.cursor()` is just instantiating a cursor class which contains no IO operation.
but it's a good practice to reuse cursor for code simplicity.

File: test/integ/aio/test_bindings_async.py [hunk 2]
```diff
@@ -0,0 +1,617 @@
...
+                    name=db_parameters["name"]
+                )
+            )
+
+
+async def test_pendulum_binding(conn_cnx, db_parameters):
+    pendulum_test = pendulum.now()
+    try:
+        async with conn_cnx() as cnx:
+            await cnx.cursor().execute(
```
[the remark this answers is not in the document]
Comment (sfc-gh-aling) [reply test/integ/aio/test_bindings_async.py]: updated

**Pull request merged**